src/platform/Darwin/inet/UDPEndPointImplNetworkFrameworkListenerGroup.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "NetworkMonitor.h"
+
+#include <inet/InetConfig.h>
+
+namespace chip {
+namespace Inet {
+namespace Darwin {
+class UDPEndPointImplNetworkFrameworkListenerGroup : public NetworkMonitor
```

Comment:
So we are ending up with UDPEndPointImplNetworkFramework multiple-inheriting, in a public way, from NetworkMonitor?

First of all, is the multiple inheritance ok?  I really hope NetworkMonitor has no virtual functions..

Second, should these be private inheritance, not public?